tensorflow_probability/python/mcmc/kernel.py

```diff
@@ -84,3 +84,25 @@ def bootstrap_results(self, init_state):  # pylint: disable=unused-argument
         `Tensor`s representing internal calculations made within this function.
     """
     return []
+
+  def copy(self, **override_parameter_kwargs):
+    """Non-destructively creates a deep copy of the kernel
+
+    Args:
+      **override_parameter_kwargs: Python String/value `dictionary` of
+        initialization arguments to override with new values.
+
+    Returns:
+      new_kernel: A `TransitionKernel` object of same type as `self`,
+        initialized with the union of self.parameters and
+        override_parameters_kwargs, with all shared keys being
+        overriden by the value in override_parameter_kwargs i.e.,
+        `dict(self.parameters, **override_parameters_kwargs)`.
+    """
+    parameters = dict(self.parameters, **override_parameter_kwargs)
+    new_kernel = type(self)(**parameters)
+    # pylint: disable=protected-access
+    new_kernel._parameters = parameters
+    # pylint: enable=protected-access
+    new_kernel = type(self)(**parameters)
```

Comment:
This line looks redundant?  Merge failure?

Reply:
Oopsies on my part; fixed